### ISSUE TITLE
Recognize a person's profile

### ISSUE THREAD

**Issue by geezz02** [habrman/FaceRecognition#23]
Hello author, when I was running the program, I found that the program could not detect my face when I turned my head over 90°, and also when I put on a mask. May I ask if there is a solution to this problem? Thanks.

**Comment by geezz02** [comment]
Hello, I have another question, when I use other people's face pictures to recognize, the program will recognize others as me, the distance is generally 0.8-0.9, and the distance is usually 0.2-0.3 when I recognize myself, may I ask how I can improve this situation?

**Comment by habrman** [comment]
>Hello author, when I was running the program, I found that the program could not detect my face when I turned my head over 90°, and also when I put on a mask. May I ask if there is a solution to this problem? Thanks.

I'm using pretrain [Multi-task Cascaded Convolutional Networks](https://kpzhang93.github.io/MTCNN_face_detection_alignment/index.html) and this particular network isn't trained on those types of faces. If you want to use something else you need to replace the use of [this](https://github.com/habrman/FaceRecognition/blob/master/detect_and_align.py) file and especially [this](https://github.com/habrman/FaceRecognition/blob/4c9516b538161e4eda5bf013ac64430051fc0956/main.py#L151) line with a differnt detector.

>Hello, I have another question, when I use other people's face pictures to recognize, the program will recognize others as me, the distance is generally 0.8-0.9, and the distance is usually 0.2-0.3 when I recognize myself, may I ask how I can improve this situation?

The default `threshold` is 1.0 but sounds like you can lower it to 0.5. You can do that by using the threshold argument here. https://github.com/habrman/FaceRecognition/blob/4c9516b538161e4eda5bf013ac64430051fc0956/main.py#L225

**Comment by geezz02** [comment]
I changed the threshold and successfully get a higher accuracy,thanks for your help and advice.

**Comment by alexeybozhchenko** [comment]
@geezz02 Did you succeed recognizing people with the mask on?

**Comment by geezz02** [comment]
> @geezz02 Did you succeed recognizing people with the mask on?

I haven't now,you might look at the author's suggestion to make some changes to the neural network.I think reducing the key points might be able to detect a person wearing a mask.

**Comment by alexeybozhchenko** [comment]
@geezz02 do you want to detect masked faces or whether person is wearing a mask?

**Comment by geezz02** [comment]
Just when a person is wearing a mask,the program can recognize the face

**Comment by habrman** [comment]
@geezz02 glad to hear that it worked. Closing the issue.
